docs/api/js/types.md

```diff
@@ -1,226 +1,303 @@
+
+## `L2ToL1Log`
+
+Interface representation of a layer 2 to layer 1 transaction log containing various fields.
+
+- `blockNumber`: `number`;
+- `blockHash`: `string`;
+- `l1BatchNumber`: `number`;
+- `transactionIndex`: `number`;
+- `shardId`: `number`;
+- `isService`: `boolean`;
+- `sender`: `string`;
+- `key`: `string`;
+- `value`: `string`;
+- `transactionHash`: `string`;
+- `logIndex`: `number`;
+
+## `Log`
+
+Interface representation of log that extends Ethers `providers.Log` and supplies the layer 1 batch number.
+
+- `l1BatchNumber`: `number`;
+
+## `MessageProof`
+
+Interface representation of message proof containing various fields.
+
+- `id`: `number`;
+- `proof`: `string[]`;
+- `root`: `string`;
+
+## `Network`
+
+Enumerated list of networks and their ids.
+
+- Mainnet = `1`
+- Ropsten = `3`
+- Rinkeby = `4`
+- Goerli = `5`
+- localhost = `9`
+
+## `PaymasterInput`
+
+Type definition for a paymaster input specified as either approval based or general.
+
+- `ApprovalBasedPaymasterInput` | `GeneralPaymasterInput`
+
+## `PaymasterParams`
+
+Type defining a paymaster by address and the bytestream input.
+
+- `paymaster`: `Address`;
+- `paymasterInput`: `BytesLike`;
+
+## `PriorityOpResponse`
+
+Interface representation of priority op response that extends [`TransactionResponse`](#transactionresponse) and adds a function that waits to commit a layer 1 transaction, including when given on optional confirmation number.
+
+- `waitL1Commit(confirmation?: number)`: `Promise<providers.TransactionReceipt>`;
+
+## `PriorityOpTree`
+
+Enumerated list of priority op tree types.
+
+- Full = `0`
+- Rollup = `1`
+
+## `PriorityQueueType`
+
+Enumerated list of priority queue types.
+
+- Deque = `0`
+- HeapBuffer = `1`
+- Heap = `2`
+
+## `Signature`
+
+0x-prefixed, hex-encoded, ECDSA signature as string.
+
+## `Token`
+
+Interface representation of token containing various fields.
+
+- `l1Address`: `Address`;
+- `l2Address`: `Address`;
+- `address`: `Address`; // backward compatible field although @deprecated in favor of l2Address 
+- `name`: `string`;
+- `symbol`: `string`;
+- `decimals`: `number`;
+
+## `TransactionDetails`
+
+Interface representation of transaction details containing various mandatory and optional fields.
+
+- `isL1Originated`: `boolean`;
+- `status`: `string`;
+- `fee`: `BigNumberish`;
+- `initiatorAddress`: `Address`;
+- `receivedAt`: `Date`;
+- `ethCommitTxHash?`: `string`;
+- `ethProveTxHash?`: `string`;
+- `ethExecuteTxHash?`: `string`;
+
+## `TransactionReceipt`
+
+Interface representation of transaction receipt that extends from `ethers.TransactionReceipt` with additional fields.
+
+- `l1BatchNumber`: `number`;
+- `l1BatchTxIndex`: `number`;
+- `logs`: `Array<Log>`;
+- `l2ToL1Logs`: `Array<L2ToL1Log>`;
+
+## `TransactionRequest`
+
+Transaction request type extended from the Ethers `providers.TransactionRequest` which adds an optional field for EIP-712 transactions.
+
+- `customData?`: `Eip712Meta`;
+
+## `TransactionResponse`
```

Comment:
i would add info that this type extends providers.TransactionResponse from ethers

Reply:
DONE